### ISSUE TITLE
Field value not set properly on first save.

### ISSUE THREAD

**Issue by alex-dna** [lekoala/silverstripe-encrypt#5]
Hi,
I'm having an issue with EncryptedDBField.
When creating a new record that has an field set to EncryptedDBField, I can see everything gets written in the database correctly, but after the first save, the fields appear blank in the CMS.
If I set the value again and re-save the record, then the value appears properly.
Here is my object:

```
<?php

use LeKoala\Encrypt\EncryptedDBField;
use LeKoala\Encrypt\HasEncryptedFields;
use SilverStripe\ORM\DataObject;

class EncryptedObject extends DataObject
{
    use HasEncryptedFields;

    private static $singular_name = "Encrypted Object";

    private static $db = [
        'Name' => EncryptedDBField::class,
        'Other' => 'Varchar(255)'
    ];

    private static $indexes = [
        'NameBlindIndex' => true,
    ];

    public function getField($field)
    {
        return $this->getEncryptedField($field);
    }

    public function setField($fieldName, $val)
    {
        return $this->setEncryptedField($fieldName, $val);
    }
}
```

Is there anything wrong with this?
I've tried to pin point where the value doesn't get set, but wasn't able to figure it out.
Thanks.

**Comment by lekoala** [comment]
@alex-dna good catch. That seems to be a regression caused by adding AAD fields. If you disable that it should be fine. My guess is that on first save, record->ID is 0, but after write, it has an id, therefore content cannot be decrypted
I have to see how to make that work properly
In the meantime, i think it's quite safe to disable aad which is just an extra security layer

**Comment by lekoala** [comment]
@alex-dna fixed in https://github.com/lekoala/silverstripe-encrypt/releases/tag/3.4.1

**Comment by alex-dna** [comment]
@lekoala Thank you so much for fixing it so quickly. Seems to work well.
I found a couple of php 8 deprecation it will fix and put up as pull request soon.

On a slightly unrelated topic: have you been successful in encrypting the FirstName and Surname field on Member?
I managed to get them encrypted by Injecting a new class with the trait and getField methods. However, that causes an issue with graphql on build...
